TWNIA2/2. Tutorials/4. 存取雲端物件儲存 (COS).md

```diff
@@ -1 +1,273 @@
+
+
+### Installtion and Settings
+
+- Accessing TWCC COS required **Access Key** and **Secret Key**. Please login to [TWCC portal](https://www.twcc.ai/) > COS to obtain it.
+- Enter the **Third Party Software** page of `Cloud Object Storage (Shared storage space between project members)` or `Private COS (private storage space)` to obtain the key information.
+
+![](https://cos.twcc.ai/SYS-MANUAL/uploads/upload_716af2cfd86e8e784e3bce0f1268254e.png)
+
+
+
+#### S3cmd
+- **Install**
+```bash
+# Using pip to install
+$ pip install s3cmd --user
+```
+- **Enter key**
+```bash
+# In here using vim as an example, you may use your familiar editor to edit ~/.s3cfg
+$ vim ~/.s3cfg
+```
+
+```bash
+# Edit content
+-----------------------
+# Setup endpoint
+host_base = cos.twcc.ai
+host_bucket = cos.twcc.ai
+use_https = True
+
+# Setup access keys
+access_key = <COS Access Key>
+secret_key = <COS Secret Key>
+-----------------------
+```
+
+#### TWCC CLI
+- **Install**
+```bash
+# Using pip to install
+$ pip3 install TWCC-CLI --user
+```
+- For **How to enter key, select projects, enter TWCC CLI environment, see [TWCC CLI document](https://man.twcc.ai/@twccdocs/twcc-cli-v05#1-3-%E9%80%B2%E5%85%A5-TWCC_CLI-%E7%92%B0%E5%A2%83%E4%B8%A6%E9%96%8B%E5%A7%8B%E4%BD%BF%E7%94%A8%E6%9C%8D%E5%8B%99)** for more information.
+
+
+<!-- #### MCLI
+- **安裝**
+
+已預先裝於台灣杉二號，使用以下指令載入工具即可。
+
+```bash
+$ module load mcli
+```
+
+- **寫入金鑰**
+
+加入金鑰，指令及參考範例如下。
+
+```bash
+$ mcli config host add twcc https://cos.twcc.ai <COS Access Key> <COS Secret Key>
+``` -->
+
+### Operation example
+
+Each example shows the operation method of using the commands `S3cmd` and `TWCC CLI` in order.
+
+- **Create a bucket named `mytwccbucket`**
+  
+```bash
+# for S3cmd
+$ s3cmd mb s3://mytwccbucket
+
+# for TWCC CLI
+$ twccli mk cos -bkt mytwccbucket
+```
+
+- **Upload file**
+
+Create a empty folder in /home directory named `myfile`
+  
+```bash
+$ touch ~/myfile
+```
+Copy file to `mytwccbucket` bucket
+  
+```bash
+# for S3cmd
+$ s3cmd put ~/myfile s3://mytwccbucket/
+
+# for TWCC CLI
+$ twccli cp cos -bkt mytwccbucket -fn myfile -sync to-cos
+```
+- **View file**
+
+Check if the upload is successful
+```bash
+# for S3cmd
+$ s3cmd ls s3://mytwccbucket/myfile
+
+# for TWCC CLI
+$ twccli ls cos -bkt mytwccbucket
+```
+- **Delte file**
+
+Delete the just uploaded myfile
+```bash
+# for S3cmd
+$ s3cmd rm s3://mytwccbucket/myfile
+
+# for TWCC CLI
+$ twccli rm cos -bkt mytwccbucket -okey myfile
+```
+- **Delete bucket**
+
+Please make sure the bucket is empty before deleting bucket
+
+```bash
+# for S3cmd
+$ s3cmd rb s3://mytwccbucket
+
+# for TWCC CLI
+$ twccli rm cos -bkt mytwccbucket
+```
+
+:::info
+<i class="fa fa-paperclip fa-20" aria-hidden="true"></i> **Note:** Other parameters and description:
+- S3cmd
+May refer using `$ s3cmd --help` command or look into [<ins>s3cmd GitHub</ins>](https://github.com/s3tools/s3cmd#simple-s3cmd-howto).
```

Comment:
May refer using `$ s3cmd --help` command or look into >

Check by entering the `$ s3cmd --help` command or learn more on [<ins>s3cmd GitHub</ins>](https://github.com/s3tools/s3cmd#simple-s3cmd-howto).